lib/graphql/client/schema/enum_type.rb

```diff
@@ -78,8 +81,13 @@ def [](value)
         def cast(value, _errors = nil)
           case value
           when String
-            raise Error, "unexpected enum value #{value}" unless @values.key?(value)
-            @values[value]
+            if @values.key?(value)
+              @values[value]
+            elsif @raise_on_unknown_enum_values
+              raise Error, "unexpected enum value #{value}"
+            else
+              "__unexpected_enum_value__"
```

Comment:
Perhaps we can introduce a class similar to `EnumValue` that captures both that it is unexpected, and what its actual JSON value is. This way a client can opt into using the raw `value`. They would likely want a way to check if it unexpected/unknown.

```suggestion
              UnexpectedEnumValue.new(value)
```

Reply:
Can we just return the raw value if `raise_on_unknown_enum_values` is set to false? Not sure of the benefit of having `UnexpectedEnumValue` class.

Reply:
> They would likely want a way to check if it unexpected/unknown.

How do you propose client code be able to understand that the value is an unexpected value from the server.

Reply:
oh right. This can be a string of value ` "__unexpected_enum_value__: '#{value}'"` ?

Reply:
This function, `cast` returns a single value, currently `String` (due to `EnumValue < String` with some extra behaviour) or `nil` (no value). The nature of being unexpected likely isn't something that should be communicated through the value itself because this may be shown to a user and would result in a different value from the underlying GraphQL representation.

Basically I am thinking a function like `unknown_enum_value?` can communicate this, without being part of the string value itself. So if a client was to just treat this as a string, it would behave normally. This won't get us the [constant definition](https://github.com/github/graphql-client/blob/2c611765e34ded9e63085b4b40a8da59d8ce8732/test/test_schema.rb#L135) or [value equality check](https://github.com/github/graphql-client/blob/2c611765e34ded9e63085b4b40a8da59d8ce8732/test/test_schema.rb#L143), but that is ok as we are expecting those values. If a client wants to do enumeration things against the returned value, they could first check `nil?` and `unknown_enum_value?` before doing `free?`/`small?`/`large?`

```ruby
class UnknownEnumValue < String
  def unknown_enum_value?
    true
  end
end

class EnumValue < String
  # ...

  def unknown_enum_value?
    false
  end
end
```